plenum/server/replica.py

```diff
@@ -965,6 +969,7 @@ def report_suspicious(reason):
             report_suspicious(Suspicions.PPR_BLS_MULTISIG_WRONG)
         else:
             self.logger.warning("Unknown PRE-PREPARE check status: {}".format(why_not))
+        self._setup_for_non_master(is_catchup=False)
```

Comment:
It looks like we should move this to the top, so that we may calculate up-to-date last_ordered based on prepared certificate BEFORE we are checking for the next Primary.

Reply:
It call there because _setup_for_non_master compare lists prePreparesPendingPrevPP and preparesWaitingForPrePrepare. But, as I undersatnd, this lists can be increase after _request_missing_three_phase_messages(). And _setup_for_non_master() was been called in the end of case PP_CHECK_NOT_NEXT before my changes, but it can be incorrect.

Reply:
Ok, I see... BTW maybe we can just rely on quorum of prepares and don't check for out-of-order PrePrepares at all...